### ISSUE TITLE
Removing the layer from the map should cancel pending HTTP requests

### ISSUE THREAD

**Issue by cosme-benito** [ptv-logistics/Leaflet.NonTiledLayer#463]
TL;DR: When you remove a NonTiledLayer from the map, it should cancel the pending HTTP requests. This can be done with something like 

```ts
_abortLoading() {
    const image = this._useCanvas
      ? this._currentCanvas._image
      : this._currentImage;

    if (image != null) {
      image.src = this.emptyImageUrl;
      image.onload = Util.falseFn;
      image.onerror = Util.falseFn;
    }
  },
```

--- 

The problem I encountered is a bit wonky but can be reliably reproduced:
- Put a breakpoint on the WMS server, preventing it from answering HTTP requests
- Add NonTiledLayer to the map
- Remove the NonTiledLayer from the map
- Add the (same) NonTiledLayer to the map again
- Remove the breakpoint on the WMS server, resume flow as normal
- The layer does not show up in on the map. If you zoom in/out it shows up.

What's currently happening:

- A single HTTP request is pending
 
![image](https://github.com/user-attachments/assets/ae901cd6-2478-49a8-8188-b459198fa9b4)

What I expected to happen:

- The first request should have been canceled
- A second request should have been fired
- This would fix the problem

![image](https://github.com/user-attachments/assets/4192577c-3c71-4157-ac7c-e16031563255)

**Comment by oliverheilig** [comment]
Hi,

your proposal looks reasonable to me. The issue is that no one actively works on the project at the moment, but feel free to create a PR.

But then the question would be if you also want a new release. It has been more than 3 years since the last release (1.0.9), we would need to check how we can roll this out then.
